test/fail_compilation/test15373.d

```diff
@@ -0,0 +1,16 @@
+/*
+TEST_OUTPUT:
+---
+fail_compilation/test15373.d(15): Error: Runtime type information is not supported for `extern(C++)` classes
+---
+*/
+
+// https://issues.dlang.org/show_bug.cgi?id=15373
+
+extern(C++) class C { }
+
+void foo()
+{
+    C c = new C();
+    auto ti = typeid(c);
```

Comment:
Can you also add a test that `typeid(C)` works ?

Reply:
Done